docs/upgrade-guide-2.0.md

```diff
@@ -0,0 +1,54 @@
+# ember-auto-import 2.0
+
+## Quick Summary
+
+- apps that are have custom webpack config will need to check that their config is compatible with webpack 5
+- apps that were adding css handling (like `css-loader`, `style-loader`, and `MiniCSSExtraPlugin`) to the webpack config must remove those, because they're now included by default for compatibility with the embroider v2 package spec.
+- apps should confirm that their deployment strategy includes all files produced under `dist` (not just the traditional expected ones like `dist/assets/your-app.js` and `dist/assets/vendor.js`)
+- addons that upgrade to ember-auto-import >= 2 will only work in apps that have ember-auto-import >= 2, so they should do their own semver major releases when they upgrade
+
+# Details
+
+### Webpack 5
+
+ember-auto-import 2.0 upgrades from webpack 4 to 5. This is a potentially breaking change for any package that directly adds to webpack config. Most apps will not experience any breakage, because most common webpack 4 configs still work in webpack 5.
+
+### Embroider v2 Addon Format support
+
+ember-auto-import 2.0 handles Embroider v2-formatted addons for apps that aren't yet using Embroider.
+
+This is potentially breaking because it adds more stuff to the webpack config that people could have been manually controlling before. Specifically:
+
+- v2 addons are allowed to import css, so we add the appropriate webpack loaders
+- v2 addons are allowed to contain inline templates, decorators, and embroider macros, so all of these are now supported by ember-auto-import
```

Comment:
This is _huuuuge!_ 😍

If I'm understanding this right, it means that we can iteratively make our individual in-house addons / engines Embroider-v2-formatted-compatible, without having to also fully migrate the host app in the same swoop, but still reap the build performance improvements by pre-building & publishing the addons / engines to a registry.

Is that correct?

And if so, are there recommended workflows for monorepo setups already? I would like to use the pre-built packages, if there are no local changes, and only live-build them, if changes were made.

Reply:
Yes, that's correct. Addons will be able to port to v2. The only requirement that will impose on apps will be that they take ember-auto-import 2.0.

> And if so, are there recommended workflows for monorepo setups already? I would like to use the pre-built packages, if there are no local changes, and only live-build them, if changes were made.

I don't have any examples yet of that kind of workflow, but I would expect that people wouldn't want to commit build artifacts to their repo, so you'd have a dev process analogous to running `tsc --watch` that could cover the whole monorepo. It's not going to be expensive, because the kind of transformations you'd do (stripping typescript types, inlining templates, generating `setComponentTemplate`) are quite light.